### PR TITLE
[NEW] Allow changing app language inside application settings

### LINKED ISSUE #872: [NEW] Allow changing app language inside application settings
If I am correct right now app uses device's language settings. It would be better user experience if user could choose which language to use regardless of device's settings.

### PR DESCRIPTION
**Pull request by artrmz**
@RocketChat/ios

Closes #872 

simulator screen shot - iphone 8 - 2018-02-28 at 07 28 50

lang2

### CODE REVIEW

File: Rocket.Chat/Controllers/Language/LanguageViewController.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = viewModel.title
+    }
+
+    @objc private func resetLanguage() {
+        UserDefaults.standard.removeObject(forKey: "AppleLanguages")
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Language/LanguageViewController.swift]: I believe this key name could be an static string. What do you think? I can see that's being used in lots of places.

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 2]
```diff
@@ -200,6 +200,7 @@
 "myaccount.settings.title" = "Configurações"; // TODO: new text: Preferences
 "myaccount.settings.contactus" = "Entre em contato";
 "myaccount.settings.license" = "Licença";
+"myaccount.settings.language" = "Language"; // TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Idioma

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 3]
```diff
@@ -208,3 +209,8 @@
 "myaccount.settings.changeicon.header" = "Você pode escolher o ícone do Rocket.Chat+ para mudar a aparência em sua home screen. Escolha o ícone abaixo que você mais gostar.";
 "myaccount.settings.changeicon.errortitle" = "Não podemos alterar seu ícone";
 "myaccount.settings.changeicon.iosversion" = "Esta funcionalidade está disponível somente para usuários com versão do iOS superior a 10.3.";
+
+// Change Language
+"myaccount.settings.language.title" = "Change Language"; // TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Alterar Idioma

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 4]
```diff
@@ -208,3 +209,8 @@
...
+"myaccount.settings.language.reset" = "Reset to Default"; // TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Restaurar o Padrão

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 5]
```diff
@@ -208,3 +209,8 @@
...
+"myaccount.settings.language.message" = "You have changed language settings. You must restart Rocket.Chat+  before the new settings will take effect."; // TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Você mudou o idioma. Você precisa fechar o aplicativo e abrir novamente para que as alterações tenham efeito.

File: Rocket.Chat/Controllers/Language/LanguageViewController.swift (hunk 1 above)
Comment (artrmz) [reply Rocket.Chat/Controllers/Language/LanguageViewController.swift]: Yeah, indeed, will fix that.

**Comment by artrmz** [comment]
@rafaelks 
1. Sure
2. Will do that :)
3. I was thinking about that but I am not sure if it'll be correct? It looks better but in some languages, it should be written with small letter although maybe should make an exception here since it's the first word in a row?
4. Good idea

**Comment by rafaelks** [comment]
@artrmz 

3. By looking into Apple's language page (Settings -> General -> Languages) you can see that all languages have the first word at least in uppercase. Let's do that. 👍

File: Rocket.Chat/Resources/de.lproj/Localizable.strings [hunk 6]
```diff
@@ -208,3 +209,8 @@
 "myaccount.settings.changeicon.header" = "Du kannst eines der unterstehenden Icons für die App wählen, welches auf deinem Homescreen angezeigt wird.";
 "myaccount.settings.changeicon.errortitle" = "Konnte App Icon nicht ändern";
 "myaccount.settings.changeicon.iosversion" = "Alternative App Icons werden nicht von iOS älter als 10.3 unterstützt.";
+
+// Change Language
+"myaccount.settings.language.title" = "Change Language"; // TODO
+"myaccount.settings.language.reset" = "Reset to Default"; // TODO
+"myaccount.settings.language.message" = "You have changed language settings. You must restart Rocket.Chat+  before the new settings will take effect."; // TODO
```
Comment (soundstorm) [inline Rocket.Chat/Resources/de.lproj/Localizable.strings]: "myaccount.settings.language" = "Sprache";

"myaccount.settings.language.title" = "Sprache ändern";
"myaccount.settings.language.reset" = "Auf Standard zurücksetzen";
"myaccount.settings.language.message" = "Die Sprache wurde geändert. Damit die Änderungen wirksam werden, muss die App neu gestartet werden.";

File: Rocket.Chat/Resources/en.lproj/Localizable.strings [hunk 7]
```diff
@@ -209,3 +210,8 @@
 "myaccount.settings.changeicon.header" = "You can change how your Rocket.Chat+ icon looks like on your device. Please choose one of the icons below, whichever you like the most.";
 "myaccount.settings.changeicon.errortitle" = "Cannot change icon";
 "myaccount.settings.changeicon.iosversion" = "Alternate application icons are not supported on iOS version below 10.3";
+
+// Change Language
+"myaccount.settings.language.title" = "Change Language";
+"myaccount.settings.language.reset" = "Reset to Default";
+"myaccount.settings.language.message" = "You have changed language settings. You must restart Rocket.Chat+  before the new settings will take effect.";
```
Comment (soundstorm) [inline Rocket.Chat/Resources/en.lproj/Localizable.strings]: Remove double space after "Rocket.Chat+" or maybe keep it general "the App"?

Comment (artrmz) [reply Rocket.Chat/Resources/en.lproj/Localizable.strings]: Removed the space. About changing it to the "the App"... I am not sure, @rafaelks what do you think?

File: Rocket.Chat/Resources/de.lproj/Localizable.strings (hunk 6 above)
Comment (artrmz) [reply Rocket.Chat/Resources/de.lproj/Localizable.strings]: Thanks! 👍

**Comment by artrmz** [comment]
@rafaelks Done. Applied all suggestions and resolved conflicts with develop.

**Comment by rafaelks** [comment]
@artrmz Looking great! I just updated our Greek file (@kmav) with some new English strings to match everything. Will merge after CI. @kmav If you want to update our Greek file again, please send us another PR after this one is merged, OK? Thank you!

**Comment by kmav** [comment]
@artrmz and @rafaelks It is very nice! I will do and make a few fixes in Greek translations too.

**Pull request merged**